lib/model/frames/submission.js

```diff
@@ -7,43 +7,38 @@
 // including this file, may be copied, modified, propagated, or distributed
 // except according to the terms contained in the LICENSE file.
 
-// SubmissionPartials are created solely by way of SubmissionPartial.fromXml
-// given a submission XML file. XML parsing work is done, and critical information
-// is extracted like the instanceId and the form version. This data is then stored
-// on a SubmissionPartial instance, which has a #createAll method that can reify
-// the temporary data into a real pair of Submission/SubmissionDef records
-// in the database.
-
 const uuid = require('uuid/v4');
-const Instance = require('./instance');
-const Option = require('../../util/option');
-const Problem = require('../../util/problem');
+const { Frame, table, readable, embedded, into } = require('../frame');
 const { consumeAndBuffer } = require('../../util/stream');
+const { resolve } = require('../../util/promise');
 const { blankStringToNull } = require('../../util/util');
 const { traverseXml, findOne, root, node, attr, text } = require('../../util/xml');
+const Option = require('../../util/option');
+const Problem = require('../../util/problem');
 
+/* eslint-disable no-multi-spaces */
 
-module.exports = Instance()(({ submissions, SubmissionPartial }) => class {
-
-  // returns a { submission, submissionDef } object.
-  createAll(form, maybeActor, deviceId = null) {
-    const actorId = Option.of(maybeActor).map((actor) => actor.id).orNull();
-    const formattedDeviceId = blankStringToNull(deviceId);
-    return submissions.create(this, form, actorId, formattedDeviceId);
-  }
-
-  // just a stub for now; we don't need this feature just yet.
-  //createVersion(form, submission, maybeActor) { throw new Error('NYI'); }
+class Submission extends Frame.define(
+  table('submissions'),
+  'id',         readable,               'formId',       readable,
```

Comment:
I think we should consider not making these two fields readable: I think `instanceId` is the id that API users should use for the submission, not `id`; we also generally don't expose `formId` (as opposed to `xmlFormId`) over the API.

Reply:
yeah that's oops